awx/main/models/notifications.py

```diff
@@ -233,14 +237,20 @@ def send_notification_templates(self, status_str):
         if notification_templates:
             if status_str == 'succeeded':
                 notification_template_type = 'success'
+            elif status_str == 'running':
+                notification_template_type = 'started'
             else:
                 notification_template_type = 'error'
-            all_notification_templates = set(notification_templates.get(notification_template_type, []) + notification_templates.get('any', []))
-            if len(all_notification_templates):
-                try:
-                    (notification_subject, notification_body) = getattr(self, 'build_notification_%s_message' % status_str)()
-                except AttributeError:
-                    raise NotImplementedError("build_notification_%s_message() does not exist" % status_str)
+            all_notification_templates = set(notification_templates.get(notification_template_type, []))
+            if status_str != 'running':
+                all_notification_templates.update(notification_templates.get('any', []))
+            try:
+                (notification_subject, notification_body) = getattr(self, 'build_notification_%s_message' % status_str)()
+            except AttributeError:
+                raise NotImplementedError("build_notification_%s_message() does not exist" % status_str)
+
+            def send_it():
                 send_notifications.delay([n.generate_notification(notification_subject, notification_body).id
                                           for n in all_notification_templates],
                                          job_id=self.id)
+            connection.on_commit(send_it)
```

Comment:
this is kind of weird to add the on_commit here as well as in those other places. I'd prefer to just just it once.

Reply:
Agreed.  @AlanCoding how do you feel about *just* using the `on_commit` _here_?

Reply:
That's fine with me, I have no opinion about which place it is put.

> until after the web request transaction is over.

 transactions defined manually within task-based actions, this is never called from a web request. But agree otherwise. As @chrismeyersfsu often points out, if you're not in a transaction (job run) then it runs right away, which is correct.

Reply:
Redundancy has been removed!